cognite/seismic/protos/v1/seismic_service.proto

```diff
@@ -47,6 +47,21 @@ service SeismicAPI {
      */
     rpc DeleteSurvey(DeleteSurveyRequest) returns (DeleteSurveyResponse) {}
 
+    /**
+     * Registers a new file in a (previously registered) survey as source for ingestion.
+     */
+    rpc RegisterSourceSegyFile (RegisterSourceSegyFileRequest) returns (RegisterSourceSegyFileResponse) {}
+
+    /**
+     * Edits a registered source file.
+     */
+    rpc EditSourceSegyFile (EditSourceSegyFileRequest) returns (EditSourceSegyFileResponse) {}
+
+    /**
+     * Unregisters a file previously registered as source for ingestion.
+     */
+    rpc UnregisterSourceSegyFile (UnregisterSourceSegyFileRequest) returns (UnregisterSourceSegyFileResponse) {}
```

Comment:
It's worth thinking a bit about the semantics of this actually. The old `DeleteFileRequest` will delete both the seismicstore and the file (if `keep_registered = false`). Should this do the same? Or should it fail if the seismicstore is still there, requiring a separate `DeleteSeismicStore` first? I think maybe the latter is wise.

Doesn't make much difference for this pr though, except for documentation.

Reply:
Based on your last comment, this is what the Rust side was doing and assuming always false.
Did not consider failing on the seismicstore existing.

I guess it matters for this PR because I could document this behavior here, if we decide to go this way.

But my two cents on this:
I believe we should not fail. My reasoning is that `SourceSegyFile` seems to me as only means for ingesting. In fact, based on the previous comments, you can edit a file post ingestion and completely mismatch one another.
Would it be a valid theoretical workflow to **register file** -> **ingest** -> **(optionally edit file and reingest)** -> **unregister file**?

Reply:
Hmm, you mean that we could allow deleting a `SourceSegyFile` leaving the ingested seismicstore in place? It's possible, but I'm not so sure it's a good idea. Seems like it would lead to confusion, without any real use case.